src/connector/src/source/cdc/external/sql_server.rs

```diff
@@ -0,0 +1,468 @@
+                .context("invalid sql server change_lsn")?,
+            commit_lsn: dbz_offset
+                .source_offset
+                .commit_lsn
+                .context("invalid sql server commit_lsn")?,
+        })
+    }
+}
+
+pub struct SqlServerExternalTable {
+    column_descs: Vec<ColumnDesc>,
+    pk_names: Vec<String>,
+}
+
+impl SqlServerExternalTable {
+    pub async fn connect(config: ExternalTableConfig) -> ConnectorResult<Self> {
+        tracing::debug!("connect to sql server");
+
+        let mut client_config = Config::new();
+
+        client_config.host(&config.host);
+        client_config.database(&config.database);
+        client_config.port(config.port.parse::<u16>().unwrap());
+        client_config.authentication(tiberius::AuthMethod::sql_server(
+            &config.username,
+            &config.password,
+        ));
+        client_config.trust_cert();
+
+        let tcp = TcpStream::connect(client_config.get_addr()).await?;
+        tcp.set_nodelay(true)?;
+
+        let mut client: tiberius::Client<Compat<TcpStream>> =
+            match tiberius::Client::connect(client_config, tcp.compat_write()).await {
+                // Connection successful.
+                Ok(client) => Ok(client),
+                // The server wants us to redirect to a different address
+                Err(Error::Routing { host, port }) => {
+                    let mut client_config = Config::new();
+
+                    client_config.host(&host);
+                    client_config.port(port);
+                    client_config.authentication(tiberius::AuthMethod::sql_server(
+                        &config.username,
+                        &config.password,
+                    ));
+
+                    let tcp = TcpStream::connect(client_config.get_addr()).await?;
+                    tcp.set_nodelay(true)?;
+
+                    // we should not have more than one redirect, so we'll short-circuit here.
+                    tiberius::Client::connect(client_config, tcp.compat_write()).await
+                }
+                Err(e) => Err(e),
+            }?;
+
+        let mut column_descs = vec![];
+        let mut pk_names = vec![];
+        {
+            // With `WHERE 1 = 0`, we only fetch the metadata (column names and types) of the table
+            let sql = Query::new(format!(
+                "SELECT * FROM {} WHERE 1 = 0",
+                SqlServerExternalTableReader::get_normalized_table_name(&SchemaTableName {
+                    schema_name: config.schema.clone(),
+                    table_name: config.table.clone(),
+                }),
+            ));
+
+            let mut stream = sql.query(&mut client).await?;
+            while let Some(item) = stream.try_next().await? {
+                match item {
+                    QueryItem::Metadata(meta) => {
+                        for col in meta.columns() {
+                            column_descs.push(ColumnDesc::named(
```

Comment:
can we get these column meta from the `INFORMATION_SCHEMA` (in the following query)? Because our pg and mysql source all rely on the information in `INFORMATION_SCHEMA` to support the auto schema mapping.

Reply:
Will do it later.